### PR TITLE
add flag to ignore local publications

### PR DESCRIPTION
**Pull request by dirk-thomas**
Required for ros2/ros1_bridge#1

Please review.

### CODE REVIEW

File: rosidl_typesupport_opensplice_cpp/resource/msg__type_support.cpp.template [hunk 1]
```diff
@@ -212,8 +216,26 @@ take__@(spec.base_type.type)(
     throw std::runtime_error("invalid ROS message pointer");
   };
 
-  @(spec.base_type.pkg_name)::@(subfolder)::@(spec.base_type.type) & ros_message = *(@(spec.base_type.pkg_name)::@(subfolder)::@(spec.base_type.type) *)untyped_ros_message;
-  convert_dds_message_to_ros(dds_messages[0], ros_message);
+  bool ignore_sample = false;
+  if (ignore_local_publications) {
+    printf("ignore_local_publications = true\n");
```
Comment (wjwwood) [inline rosidl_typesupport_opensplice_cpp/resource/msg__type_support.cpp.template]: Should this be left in?

**Comment by wjwwood** [comment]
lgtm

File: rosidl_typesupport_opensplice_cpp/resource/msg__type_support.cpp.template (hunk 1 above)
Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_cpp/resource/msg__type_support.cpp.template]: Oh no, removed it.

**Comment by esteve** [comment]
+1

**Comment by tfoote** [comment]
+1

**Pull request merged**